### PR TITLE
Draft PR: Cleaning the code base

### PR DESCRIPTION
**Pull request by rach-id**

### CODE REVIEW

**Comment by rach-id** [comment]
What this does:
- Change `com` to `org` in package names.
- Remove standalone functions
- Add a `LedgerConfiguration` class
- Fixes specifying `host` and `port`

File: src/main/kotlin/org/epirus/local/cli/StartCmd.kt [hunk 1]
```diff
@@ -10,38 +10,64 @@
...
             description = ["specify the directory of the output configuration"],
             defaultValue = ".")
     var directory: String = "."
 
-    // FIXME: when specified => network permissions required
     @Option(names = ["-p", "--port"],
             description = ["specify the port to run the client on"],
             hidden = true)
-    var port: Int = 8080
+    var cliPort: Int = 8080
```
Comment (iikirilov) [inline src/main/kotlin/org/epirus/local/cli/StartCmd.kt:43]: change these back

File: src/main/kotlin/org/epirus/local/cli/StartCmd.kt [hunk 2]
```diff
@@ -10,38 +10,64 @@
...
+
+    @KtorExperimentalAPI
     override fun call(): Int {
-        embeddedServer(Netty, commandLineEnvironment(arrayOf())).start()
+
+        val ledgerConfiguration = LedgerConfiguration(
+                directory = directory)
+        val env = applicationEngineEnvironment {
+            connector {
+                host = cliHost
```
Comment (iikirilov) [inline src/main/kotlin/org/epirus/local/cli/StartCmd.kt:58]:
```suggestion
                host = this.host
```

??? maybe not

File: src/main/kotlin/org/epirus/local/ledger/LocalLedger.kt [hunk 3]
```diff
@@ -23,23 +23,32 @@ import org.web3j.evm.Configuration
...
 import java.math.BigInteger
 import java.net.URL
-import java.util.stream.Collectors
 
-class LocalLedger(val accounts: List<Account> = emptyList(), val genesisPath: String) {
+class LocalLedger(private val ledgerConfiguration: LedgerConfiguration) {
 
     private val embeddedEthereum: EmbeddedEthereum
+    private val accounts = ledgerConfiguration.accounts
+    private val genesis = ledgerConfiguration.genesis
```
Comment (iikirilov) [inline src/main/kotlin/org/epirus/local/ledger/LocalLedger.kt]: remove

File: src/main/kotlin/org/epirus/local/cli/LoadCmd.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+@Command(name = "load",
+        description = ["Load a previous configuration for epirus-local client",
+            """Example: epirus-local load -g=folder/genesis.json"""])
+class LoadCmd : Callable<Int> {
+
+    @CommandLine.Option(names = ["-g", "--genesis"],
+            description = ["specify the genesis file"],
+            defaultValue = "./genesis.json")
+    var genesis: String = "./genesis.json"
```
Comment (iikirilov) [inline src/main/kotlin/org/epirus/local/cli/LoadCmd.kt]:
```suggestion
    var genesisFilePath: String = "./genesis.json"
```

File: src/main/kotlin/org/epirus/local/cli/LoadCmd.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    private lateinit var client: NettyApplicationEngine
+
+    @KtorExperimentalAPI
+    override fun call(): Int {
+
+        val ledgerConfiguration = LedgerConfiguration(
+                genesis = genesis)
+        val env = applicationEngineEnvironment {
+            connector {
+                host = cliHost
```
Comment (iikirilov) [inline src/main/kotlin/org/epirus/local/cli/LoadCmd.kt:57]:
```suggestion
                host = this.host
```

File: src/main/kotlin/org/epirus/local/ledger/LocalLedger.kt (hunk 3 above)
Comment (iikirilov) [reply src/main/kotlin/org/epirus/local/ledger/LocalLedger.kt]: you can call directly `ledgerConfiguration.<field>` directly

File: src/main/kotlin/org/epirus/local/ledger/LocalLedger.kt [hunk 6]
```diff
@@ -23,23 +23,32 @@ import org.web3j.evm.Configuration
...
 
     private fun loadConfig(path: String): Configuration {
+        if (!accounts.isNullOrEmpty()) {
+            logger.info("""Starting ledger with generated genesis file: $genesis""")
+            logger.info("chainID = 1")
+            accounts.forEach { t -> logger.info("Account: ${t.address} created with 100 eth and private key: ${t.privateKey}") }
+        } else {
+            logger.info("""-> Starting ledger with genesis file: $genesis""")
+        }
         return Configuration(Address("0x0"), 0, URL("file:$path"))
```
Comment (iikirilov) [inline src/main/kotlin/org/epirus/local/ledger/LocalLedger.kt:50]: open issue in web3j-evm to allow creating config with no funded account

File: src/test/kotlin/org/epirus/local/utils/Folder.kt [hunk 7]
```diff
@@ -10,21 +10,19 @@
...
     fun tempBuildFolder(): File {
         val tmpTestLocation = File(
-                String.join(
-                        File.separator,
+                arrayOf(
                         "build",
                         "tmp",
                         "testing",
-                        java.lang.Long.toString(System.currentTimeMillis())))
+                        java.lang.Long.toString(System.currentTimeMillis())).joinToString(File.separator))
```
Comment (iikirilov) [inline src/test/kotlin/org/epirus/local/utils/Folder.kt]:
```suggestion
                      System.currentTimeMillis().toString()
```

**Review by iikirilov — APPROVED** [review]
LGTM - a few things to clean-up

File: src/main/kotlin/org/epirus/local/cli/StartCmd.kt (hunk 2 above)
Comment (rach-id) [reply src/main/kotlin/org/epirus/local/cli/StartCmd.kt:58]: not working in both ways

File: src/main/kotlin/org/epirus/local/cli/LoadCmd.kt (hunk 5 above)
Comment (rach-id) [reply src/main/kotlin/org/epirus/local/cli/LoadCmd.kt:57]: not working

**Pull request merged**